apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -78,4 +115,83 @@ defmodule Aecore.Txs.Pool.Worker do
     {:reply, tx_pool, %{}}
   end
 
+  @doc """
+  A function that adds a merkle proof for every single transaction
+  """
+
+  @spec add_proof_to_txs(list()) :: list()
+  def add_proof_to_txs(user_txs) do
+    for tx <- user_txs do
+      block = get_block_by_txs_hash(tx.txs_hash)
+      tree  = BlockValidation.build_merkle_tree(block.txs)
+      key   = :erlang.term_to_binary(
```

Comment:
`:erlang.term_to_binary` should also be part `TxData.hash_tx()`.  If you would run dialyzer you would see that the spec of `TxData.hash_tx()` (which is a spec for correct implementation) is different then the actual implementation.